packages/core/test/invoke.test.ts

```diff
@@ -2219,9 +2237,12 @@ describe('invoke', () => {
     it('should work with a behavior', (done) => {
       const countBehavior: Behavior<EventObject, number> = {
         transition: (count, event) => {
-          if (event.type === 'INC') {
+          // TODO: all behaviors receive SCXML.Event objects,
+          // make sure this is clear in the docs
```

Comment:
question is - should they? it complicates our internals and the public interface for a questionable benefit

I'm also not sure if this note in the comment is actually correct. Either it's outdated or the declared types don't reflect this comment.

Reply:
> question is - should they? it complicates our internals and the public interface for a questionable benefit

Yes; SCXML.Event objects hold `origin` metadata.